src/app/share-report/share-report.component.ts

```diff
@@ -118,6 +118,8 @@ export class ShareReportComponent implements AfterViewInit {
     },
   ];
   selectedTwitterActionOptionIndex = 0;
+  selectedTwitterAction =
+    this.twitterActionOptions[this.selectedSaveOptionIndex].action;
```

Comment:
I think this should be selectedTwitterActionOptionIndex instead of selectedSaveOptionIndex